hydro_cli/src/core/hydroflow_crate/mod.rs

```diff
@@ -38,7 +38,8 @@ pub struct HydroflowCrate {
     /// Configuration for the ports that this service will listen on a port for.
     port_to_bind: HashMap<String, ServerStrategy>,
 
-    built_binary: Option<JoinHandle<Result<BuiltCrate>>>,
+    building_binary: Option<JoinHandle<Result<BuiltCrate>>>,
+    built_binary: Option<BuiltCrate>,
```

Comment:
maybe using something like `MaybeDone` would be cleaner with these two fields
https://docs.rs/futures/latest/futures/future/fn.maybe_done.html

Reply:
Or async once cell maybe https://docs.rs/async-once-cell/latest/async_once_cell/struct.OnceCell.html